fermenter-mda/src/main/java/org/technologybrewery/fermenter/mda/element/ExpandedProfile.java

```diff
@@ -112,6 +112,10 @@ private void dereferenceProfilesReferences(Map<String, ExpandedProfile> profiles
                     messageTracker.addWarningMessage("Profile '" + getName() + "' cannot include itself!");
 
                 } else {
+                    ExpandedProfile profile = profiles.get(nameOnlyProfile.getName());
+                    if (profile != null) {
+                        addReferencedProfile(profile);
+                    }
```

Comment:
I: Seems like following profile referenced entities doesn't even get added nor used as part of dereferencing.
protected Map<String, ExpandedProfile> includes = new HashMap<>();

    public void addReferencedProfile(ExpandedProfile profile) {
        includes.put(profile.getName(), profile);
    }
Added logic to fix as this is needed for unit testing.